src/org/openstreetmap/josm/gui/dialogs/properties/PropertiesDialog.java

```diff
@@ -304,6 +316,21 @@ public PropertiesDialog() {
         editHelper.loadTagsIfNeeded();
 
         TaggingPresets.addListener(this);
+
+        PROP_PREVIEW_ON_HOVER.addListener(hoverPreviewPropListener);
+    }
+
+    private class HoverPreviewPropListener implements ValueChangeListener<Boolean> {
+        @Override
+        public void valueChanged(ValueChangeEvent<? extends Boolean> e) {
+            if (e.getProperty().get() && isDialogShowing() && !haveMouseMotionListener) {
+                MainApplication.getMap().mapView.addMouseMotionListener(PropertiesDialog.this);
+                haveMouseMotionListener = true;
+            } else if (!e.getProperty().get() && haveMouseMotionListener) {
+                MainApplication.getMap().mapView.removeMouseMotionListener(PropertiesDialog.this);
+                haveMouseMotionListener = false;
+            }
+        }
```

Comment:
This should probably be down near the bottom of the file, with the other private class.

Reply:
I originally put it up there to be close to where it's actually being used, but I moved it to the bottom now.